compiler/rustc_query_system/src/query/job.rs

```diff
@@ -136,18 +136,16 @@ impl QueryJobId {
 
     #[cold]
     #[inline(never)]
-    pub fn try_find_layout_root(
-        &self,
-        query_map: QueryMap,
-        layout_of_kind: DepKind,
-    ) -> Option<(QueryJobInfo, usize)> {
-        let mut last_layout = None;
-        let mut current_id = Some(*self);
-        let mut depth = 0;
+    pub fn try_find_dep_kind_root(&self, query_map: QueryMap) -> Option<(QueryJobInfo, usize)> {
```

Comment:
This is now always Some

```suggestion
    pub fn try_find_dep_kind_root(&self, query_map: QueryMap) -> (QueryJobInfo, usize) {
```